ssm.go

```diff
@@ -10,20 +10,19 @@ import (
 type DefaultSSMConnector struct{}
 
 func (c DefaultSSMConnector) FetchParameters(paths []string, recursive bool, retries int) (map[string]string, error) {
-	params := map[string]string{}
-
-	sess, err := session.NewSession()
+	client, err := newSSMClient(retries)
 	if err != nil {
-		return params, errors.Wrap(err, "failed to start session")
+		return nil, err
 	}
+	return c.fetchParameters(client, paths, recursive)
+}
 
-	// config.MaxRetries is defaults to -1
-	if retries <= 0 {
-		retries = -1
+func (c DefaultSSMConnector) fetchParameters(client *ssm.SSM, paths []string, recursive bool) (map[string]string, error) {
```

Comment:
plan1

Align naming rule (I prefer)

- fetchParametersByPaths
- fetchParametersByNames

plan2

Fit to aws-sdk-go

- fetchParametersByPaths
- fetchParameters

Reply:
fixed by d20f403.